pkg/action/list.go

```diff
@@ -150,6 +154,15 @@ func (l *List) Run() ([]*release.Release, error) {
 		return nil, err
 	}
 
+	if l.AllNamespaces {
+		// For Empty Namespace the default behavior is to get all namespaces
+		clonedCfg, err := l.cfg.CloneWithNewNamespace("")
+		if err != nil {
+			return nil, err
+		}
+		l.cfg = clonedCfg
```

Comment:
```go
l.cfg = clonedCfg
```
Looks good! I don't know if this matters but this approach potentially breaks reusing an list action. Do you think that's a problem?

Reply:
I agree with @AleksandarFaraj -- since `l.cfg` is only used [#L175](https://github.com/helm/helm/pull/11674/files#diff-957d45c5563ffa9a681a8aa0bb489b1297b8e8afed5858d7bfd10fce32c94ad2R175), it should be simple to do something like:

```
cfg := l.cfg
if l.AllNamespaces {
    cfg, err := l.cfg.CloneWithNewNamespace("")
    ...
```

And below do:
```
results, err := cfg.Releases.List(func(rel *release.Release) bool {
```

Reply:
https://github.com/helm/helm/blob/55f35060cbc662df056801fafc5c33ce1e4a5ad4/pkg/action/list.go#L152-L157
Would it be better to make this declaration before the reachable check such that it checks the configuration it cloned over the previous one?

Reply:
I may be wrong but aren't you declaring cfg twice this way ?

```
	cfg := l.cfg
	if l.AllNamespaces {
		cfg, err := l.cfg.CloneWithNewNamespace("")
		// For Empty Namespace the default behavior is to get all namespaces
		if err != nil {
			return nil, err
		}
	}
```
So below:
```
results, err := cfg.Releases.List(func(rel *release.Release) bool {
```
Will this work ?

Reply:
You are right, I wrote this app to test it. Just use the clonedCfg and then cfg = clonedCfg

```go
package main

func main() {
	a := true
	b := 5
	if a {
		b, e := r() // b declared but not used (Seems to be local to the if-statement).
		if e != nil {
			print(e)
		}
	}
	println(b)
}
func r() (int, error) {
	return 5, nil
}
```

Reply:
> I may be wrong but aren't you declaring cfg twice this way ?

Yep, I was trying to keep my example concise (perhaps too concise :) ). I've updated it to be more correct.